### PR TITLE
AddBulletCommand modification and some minor modification

### LINKED ISSUE #271: Typo in Entrybook
The `addEntry` method is defined as `addEnty`

image

### LINKED ISSUE #319: Modify AddBullet Command
1. Disallow addition of duplicated bullets
2. refactor AddBullet Command
3. Complete test cases for addBullet command

### PR DESCRIPTION
**Pull request by jhengy**
fixes #319 
fixes #271 
AddBullet Command modification
- AddBullet Command check for duplicateBullet
- refactor AddBulletCommand
- Complete tests
minor modification
- refactor EntryBook#addEnty
- remove "to be modified" comment for ModelManager#hasEntry()

### CODE REVIEW

File: src/main/java/seedu/address/logic/commands/AddBulletCommand.java [hunk 1]
```diff
@@ -28,6 +28,7 @@
             + "attained Best Financial Hack Award";
 
     public static final String MESSAGE_ADDBULLET_SUCCESS = "Added Bullet : %1$s";
+    public static final String MESSAGE_ADDBULLET_DUPLICATE_BULLET = "The bullet to add already exists";
```
Comment (anubh-v) [inline src/main/java/seedu/address/logic/commands/AddBulletCommand.java]: Minor suggestion: I think we can say "this bullet already exists".

File: src/main/java/seedu/address/model/entry/EntryDescription.java [hunk 2]
```diff
@@ -45,9 +45,17 @@ public EntryDescription(EntryDescription toCopy) {
...
         checkArgument(isValidBullet(bullet), MESSAGE_ENTRYDESC_CONSTRAINTS);
+
         descriptionList.add(bullet);
     }
 
+    /**
+     * checks if a bullet is contained in the EntryDescription.
+     */
+    public boolean contains(String bullet) {
+        return descriptionList.stream().anyMatch(s -> s.equals(bullet));
```
Comment (anubh-v) [inline src/main/java/seedu/address/model/entry/EntryDescription.java]: Just wondering - can we use the `List.contains` method here? May be simpler

File: src/test/java/seedu/address/logic/commands/AddBulletCommandTest.java [hunk 3]
```diff
@@ -32,6 +35,15 @@
                                                    new Awareness());
     private CommandHistory commandHistory = new CommandHistory();
 
+    /**
+     * Get an entry book with one typical entry with description.
+     */
+    private EntryBook getEntrybookWithDescription() {
+        EntryBook entryBook = new EntryBook();
+        entryBook.addEnty(WORK_FACEBOOK_WITH_DESC);
```
Comment (anubh-v) [inline src/test/java/seedu/address/logic/commands/AddBulletCommandTest.java]: Will be good if you can fix the typo in the Entrybook --> `entryBook.addEnty`

File: src/main/java/seedu/address/model/entry/EntryDescription.java (hunk 2 above)
Comment (jhengy) [reply src/main/java/seedu/address/model/entry/EntryDescription.java]: Modified

File: src/main/java/seedu/address/logic/commands/AddBulletCommand.java (hunk 1 above)
Comment (jhengy) [reply src/main/java/seedu/address/logic/commands/AddBulletCommand.java]: Modified

File: src/test/java/seedu/address/logic/commands/AddBulletCommandTest.java (hunk 3 above)
Comment (jhengy) [reply src/test/java/seedu/address/logic/commands/AddBulletCommandTest.java]: will do it in a separate PR

Comment (jhengy) [reply src/test/java/seedu/address/logic/commands/AddBulletCommandTest.java]: Actually, I will just do it here

**Pull request merged**